config/dynamic.c

```diff
@@ -526,6 +539,17 @@ run_dynamic_config(const char *server_path, struct dynamic_config *dy_conf)
 		goto free_sock;
 	}
 
+	/*
+	 * TODO Add support for the dynamic configuration running as Grantor.
+	 */
+	if (gk_conf != NULL) {
+		gk_conf_hold(gk_conf);
+		dy_conf->gk = gk_conf;
+	} else {
+		ret = -1;
+		goto free_sock;
+	}
+
 	ret = launch_at_stage3("dynamic_conf",
 		dyn_cfg_proc, dy_conf, dy_conf->lcore_id);
 	if (ret < 0)
```

Comment:
If `launch_at_stage3()` fails, you can't go to label `free_sock` because we may have called `gk_conf_hold(gk_conf)`; we need a new label for to deal with this possibility.